test/data/mini_example/mini_example.sam

```diff
@@ -0,0 +1,69 @@
+read027	0	chr1	126	60	4S46M	*	0	0	ATTTGGATCTTGACTCTGGAAAACTTTTAACGCCGGGAATCGGTAGTCCT	*	AS:i:46	NM:i:0	SA:Z:chr1,94,+,4M46S,60,4,0;
+read028	0	chr1	139	60	50M	*	0	0	GGAAAACTTTTAACGCCGGGAATCGGTAGTCCTTTCGCGGGGATATATTT	*	AS:i:50	NM:i:0
+read029	0	chr1	146	60	43M7S	*	0	0	TTTTAACGCCGGGAATCGGTAGTCCTTTCGCGGGGATATATTTATATATT	*	AS:i:43	NM:i:0	SA:Z:chr1,181,+,43S7M,60,7,0;
+read029	2056	chr1	146	60	35M8I7M	*	0	0	TTTTAACGCCGGGAATCGGTAGTCCTTTCGCGGGGATATATTTATATATT	*	AS:i:38	NM:i:8
+read030	0	chr1	147	60	42M8S	*	0	0	TTTAACGCCGGGAATCGGTAGTCCTTTCGCGGGGATATATTTATATATTT	*	AS:i:42	NM:i:0	SA:Z:chr1,181,+,42S8M,60,8,0;
+read030	2056	chr1	147	60	34M8I8M	*	0	0	TTTAACGCCGGGAATCGGTAGTCCTTTCGCGGGGATATATTTATATATTT	*	AS:i:38	NM:i:8
+read031	0	chr1	151	60	30M8I12M	*	0	0	ACGCCGGGAATCGGTAGTCCTTTCGCGGGGATATATTTATATATTTTACG	*	AS:i:38	NM:i:8
+read031	2056	chr1	151	60	38M8I4M	*	0	0	ACGCCGGGAATCGGTAGTCCTTTCGCGGGGATATATTTATATATTTTACG	*	AS:i:38	NM:i:8
+read032	0	chr1	180	60	1M8I41M	*	0	0	GATATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAAT	*	AS:i:38	NM:i:8
+read032	2056	chr1	180	60	9M8I33M	*	0	0	GATATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAAT	*	AS:i:38	NM:i:8
+read029	2048	chr1	181	60	43S7M	*	0	0	TTTTAACGCCGGGAATCGGTAGTCCTTTCGCGGGGATATATTTATATATT	*	AS:i:7	NM:i:0
+read030	2048	chr1	181	60	42S8M	*	0	0	TTTAACGCCGGGAATCGGTAGTCCTTTCGCGGGGATATATTTATATATTT	*	AS:i:8	NM:i:0
+read033	0	chr1	181	60	8S42M	*	0	0	ATATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATA	*	AS:i:42	NM:i:0	SA:Z:chr1,181,+,8M42S,60,8,0;
+read033	2048	chr1	181	60	8M42S	*	0	0	ATATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATA	*	AS:i:8	NM:i:0
+read033	2056	chr1	181	60	8M8I34M	*	0	0	ATATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATA	*	AS:i:38	NM:i:8
+read034	0	chr1	181	60	7S43M	*	0	0	TATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAG	*	AS:i:43	NM:i:0	SA:Z:chr1,182,+,7M43S,60,7,0;
+read035	0	chr1	181	60	1S49M	*	0	0	TATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAGGTCTCG	*	AS:i:49	NM:i:0
+read036	0	chr1	181	60	50M	*	0	0	ATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAGGTCTCGG	*	AS:i:50	NM:i:0
+read034	2056	chr1	182	60	7M8I35M	*	0	0	TATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAG	*	AS:i:38	NM:i:8
+read034	2048	chr1	182	60	7M43S	*	0	0	TATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAG	*	AS:i:7	NM:i:0
+read035	2056	chr1	188	60	1M8I41M	*	0	0	TATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAGGTCTCG	*	AS:i:38	NM:i:8
+read037	0	chr1	222	60	45M20D5M	*	0	0	AGGTCTCGGTTGCCAACTGATCGTACCAAATATTTCTGCGGGGCTTCGGC	*	AS:i:46	NM:i:20
+read038	0	chr1	223	60	44M20D6M	*	0	0	GGTCTCGGTTGCCAACTGATCGTACCAAATATTTCTGCGGGGCTTCGGCT	*	AS:i:46	NM:i:20
+read039	0	chr1	251	60	16M17I16M17D1M	*	0	0	ATATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTAC	*	AS:i:24	NM:i:34
+read039	2056	chr1	251	60	16M20D13M20I1M	*	0	0	ATATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTAC	*	AS:i:20	NM:i:40
+read039	2056	chr1	251	60	16M16D13I4M13D16I1M	*	0	0	ATATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTAC	*	AS:i:3	NM:i:38
```

Comment:
A string with a deletion immediately following an insertion or vice versa (e.g. 16M16D13I4M13D16I1M) is unrealistic because a mismatch is probably cheaper than deletion+insertion. In the specific example of read039 it's most likely that the aligner splits the read into five pieces and aligns them individually.

Reply:
aah okay, I was wondering, how to manage these things. I will try to split it.

Reply:
At the moment:
16M16D13I4M13D16I1M
```
ref: ...CAA ATATTTCTGCGGGGCT CGCCCCTCCGCGATTA ------13----- AGAG TCGGCTAACGGTT -------16------- CAC...
            |||||||16|||||||                                ||4|                                |
read:       ATATTTCTGCGGGGCT -------16------- TCGGCTAACGGTT AGAG ------13----- CGCCCCTCCGCGATTA C
```
into
```
ref: ...CAA ATATTTCTGCGGGGCT CGCCCCTCCGCGATTA AGAG TCGGCTAACGGTT CAC...
            |||||||16|||||||                         
16M34S:       ATATTTCTGCGGGGCT TCGGCTAACGGTT AGAG CGCCCCTCCGCGATTA C

ref: ...CAA ATATTTCTGCGGGGCT CGCCCCTCCGCGATTA AGAG TCGGCTAACGGTT CAC...
                                                   ||||||13|||||                         
16S13M21S:                        ATATTTCTGCGGGGCT TCGGCTAACGGTT AGAG CGCCCCTCCGCGATTA C

ref: ...CAA ATATTTCTGCGGGGCT CGCCCCTCCGCGATTA AGAG TCGGCTAACGGTT CAC...
                                              ||4|                        
29S4M17S:      ATATTTCTGCGGGGCT TCGGCTAACGGTT AGAG CGCCCCTCCGCGATTA C

ref:                  ...CAA ATATTTCTGCGGGGCT CGCCCCTCCGCGATTA AGAG TCGGCTAACGGTT CAC...
                                              |||||||16|||||||                        
33S16M1S: ATATTTCTGCGGGGCT TCGGCTAACGGTT AGAG CGCCCCTCCGCGATTA C
```
I hope thats what you meant. :)

Reply:
Yes, that's precisely what I meant. Thanks! :)